### DISCUSSION TITLE
Flashing Micropython 1.21 to an ESP-01m not possible

### DISCUSSION THREAD

**Discussion by besi** [micropython/micropython#13323] [ESP8266] [answered]
I was not able to flash my ESP-01m with the latest micropython version 1.21. 
As soon as I did a fallback onto 1.20 I could successfully flash my device and get a REPL.

EXPECTED BEHAVIOUR

I flash the device and get a REPL when connecting to it via TTL cable.

ACTUAL BEHAVIOUR

I can flash the device and "Thonny" tells me that the flashing was complete. However when I disable the boot loader mode and reset the device the device fails to respond.

I did use the latest Thonny 4.1.4 which uses ESPTool in the background.

I quite like the [form factor](https://github.com/besi/kicad-esp-01m) of the ESP-01m so I would hope that micropython continues to work on that device.

**Comment by robert-hh** [comment, marked as answer]
Try to flash the firmware with other flash modes, like DOUT.

**Reply by besi** [reply]
That did the trick along with flash size detect. Thanks very much

**Reply by besi** [reply]
That did the trick along with flash size detect. Thanks very much

**Reply by besi** [reply]
I still find it problematic that Esptool let’s me flash the device all the way only then to let me realise that I need to change some of the settings. I wonder if that is due to hardware limitations or if it could be fixed in software.

**Reply by Josverl** [reply]
Perhaps you can add to the issue reported on the esptool repo : https://github.com/espressif/esptool/issues/816
I see that you started this, but that would be the best place to get the answers you are looking for.
